wrangler-core/src/main/java/io/cdap/functions/Logical.java

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Copyright © 2020 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.functions;
+
+/**
+ * The collection of logical functions that operate on bit.
+ */
+public final class Logical {
+
+  /**
+   * Don't let anyone instantiate this class.
+   */
+  private Logical() {}
+
+  /**
+   * Bitwise 'AND' operation of two numbers.
+   *
+   * @param num1 first number.
+   * @param num2 second number.
+   * @return a result of bitwise 'AND'.
+   */
+  public static long BitAnd(long num1, long num2) {
+    return num1 & num2;
+  }
+
+  /**
+   * Bitwise 'OR' operation of two numbers.
+   *
+   * @param num1 first number.
+   * @param num2 second number.
+   * @return a result of bitwise 'OR'
+   */
+  public static long BitOr(long num1, long num2) {
+    return num1 | num2;
+  }
+
+  /**
+   * Bitwise 'XOR' operation of two numbers.
+   *
+   * @param num1 first number.
+   * @param num2 second number.
+   * @return a result of bitwise 'XOR'
+   */
+  public static long BitXor(long num1, long num2) {
+    return num1 ^ num2;
+  }
+
+  /**
+   * Returns the complement of the logical value of an expression. If the value of expression is true,
+   * the Not function returns a value of false (0). If the value of expression is false, the NOT function
+   * returns a value of true (1). A numeric expression that evaluates to 0 is a logical value of false.
+   * A numeric expression that evaluates to anything else, other than the null value, is a logical true.
+   * An empty string is logically false. All other string expressions, including strings that include
+   * an empty string, spaces, or the number 0 and spaces, are logically true.
+   *
+   * @param exp expression result.
+   * @return 1 or 0.
+   */
+  public static short Not(double exp) {
+    if (exp < 0) {
+      return 0;
+    }
+    return 1;
+  }
+
+  /**
+   * Returns the complement of the logical value of an expression. If the value of expression is true,
+   * the Not function returns a value of false (0). If the value of expression is false, the NOT function
+   * returns a value of true (1). A numeric expression that evaluates to 0 is a logical value of false.
+   * A numeric expression that evaluates to anything else, other than the null value, is a logical true.
+   * An empty string is logically false. All other string expressions, including strings that include
```

Comment:
behavior doesn't match the description (empty string is not checked)

Reply:
Fixed